blocks/bricks/base.py

```diff
@@ -21,6 +24,48 @@ def create_unbound_method(func, cls):
 property_ = property
 
 
+class Parameters(MutableSequence):
+    """Behaves exactly like a list, but annotates the variables."""
+    def __init__(self, brick, params):
+        self.brick = brick
+        self._params = []
+        for param in params:
+            self.append(param)
+
+    def __repr__(self):
+        return repr(self._params)
+
+    def __getitem__(self, key):
+        return self._params.__getitem__(key)
+
+    def _annotate(self, value):
+        """Annotates the variable.
+
+        Raises
+        ------
+        ValueError
+            If the parameter isn't a shared variable or ``None``.
+
+        """
+        if isinstance(value, Variable):
+            add_role(value, PARAMETER)
```

Comment:
Just to make sure we're on the same page: this line means that the substituted parameters, which typically are not shared variables, can also bear the PARAMETER role, which in turn means that the reliable way to extract all trainable parameters is to request all shared variables with PARAMETER role. Am I right?